### PR TITLE
Add all files from saddle-token repo

### PR DESCRIPTION
**Pull request by hammeiam**
- deployments
- contracts
- interfaces
- libraries
- tests
- added aliased deps to maintain versions
- added some testUtils helper functions 

I have no idea how to confirm that no bytecode was changed, but that'll be necessary before continuing

### CODE REVIEW

File: deploy/mainnet/304_deploy_MiniChef.ts [hunk 1]
```diff
@@ -7,84 +7,86 @@ import { ethers } from "hardhat"
...
-  const TOTAL_LM_REWARDS = BIG_NUMBER_1E18.mul(12_500_000)
-  // 6 months (24 weeks)
-  const lmRewardsPerSecond = TOTAL_LM_REWARDS.div(6 * 4 * 7 * 24 * 3600)
+    // Total LM rewards is 30,000,000 but only 12,500,000 is allocated in the beginning
+    const TOTAL_LM_REWARDS = BIG_NUMBER_1E18.mul(12_500_000)
+    // 6 months (24 weeks)
+    const lmRewardsPerSecond = TOTAL_LM_REWARDS.div(6 * 4 * 7 * 24 * 3600)
 
-  const chainId = await getChainId()
+    const chainId = await getChainId()
```
Comment (hammeiam) [inline deploy/mainnet/304_deploy_MiniChef.ts:29]: unused?

File: test/retroactiveVesting.ts [hunk 2]
```diff
@@ -86,7 +87,10 @@ describe("Retroactive Vesting", () => {
         "RetroactiveVesting",
       )) as RetroactiveVesting
 
-      await dummyToken.mint(retroactiveVesting.address, BigNumber.from(600000))
+      await dummyToken.mint(
+        retroactiveVesting.address,
+        BIG_NUMBER_1E18.mul(1_000_000),
```
Comment (hammeiam) [inline test/retroactiveVesting.ts:92]: big number/brain

File: test/retroactiveVesting.ts [hunk 3]
```diff
@@ -250,23 +284,23 @@ describe("Retroactive Vesting", () => {
       // 1 year since start
       await setTimestamp(startTimestamp + 52 * WEEK)
       expect(await retroactiveVesting.vestedAmount(deployerAddress)).to.be.eq(
-        50000,
+        "49999995230463980463981",
```
Comment (hammeiam) [inline test/retroactiveVesting.ts:287]: close enough

Comment (penandlim) [reply test/retroactiveVesting.ts:287]: timestamp stuff is always messy :P

**Review by penandlim — APPROVED** [review]

**Comment by penandlim** [comment]
Need to confirm bytecodes didnt change before merging but code wise lgtm

**Pull request merged**